src/sp/it/pl/audio/playlist/PlaylistReader.kt

```diff
@@ -0,0 +1,91 @@
+package sp.it.pl.audio.playlist
+
+import mu.KotlinLogging
+import sp.it.pl.audio.Item
+import sp.it.pl.audio.SimpleItem
+import sp.it.pl.util.dev.fail
+import sp.it.pl.util.dev.throwIfFxThread
+import sp.it.pl.util.file.div
+import sp.it.pl.util.file.hasExtension
+import sp.it.pl.util.file.parentDir
+import sp.it.pl.util.functional.net
+import java.io.File
+import java.io.IOException
+import java.net.URI
+import java.net.URISyntaxException
+import java.nio.charset.Charset
+
+private const val EXTM3U = "#EXTM3U"
+private const val EXTINF = "#EXTINF"
+private val logger = KotlinLogging.logger { }
+
+@Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
+fun readPlaylist(file: File): List<Item> {
+    throwIfFxThread()
+
+    val location = file.parentDir!!
+    val encoding = when {
+        file hasExtension "m3u" -> Charset.defaultCharset()
+        file hasExtension "m3u8" -> Charsets.UTF_8
+        else -> fail { "File=$file is not a supported playlist file" }
+    }
+
+    file.useLines(encoding) { lines ->
+        var isFirstLine = true
+        var isExtended = false
+        return lines
+                .onEach { if (isFirstLine && EXTM3U==it) isExtended = true }
+                .onEach { isFirstLine = false }
```

Comment:
Indeed. If you know a better way tell me. Honestly we do not need this, because extended m3u's metadata are useless for us anyway, but I wanted to leave that as self documenting code. I can remove it.

Your call.

Reply:
ok

Reply:
Removed completely